lib/wickedbase.pm

```diff
@@ -1319,7 +1319,6 @@ sub wait_for_background_process {
         die("wait_for_background_process() failed, process $pid wasn't ready yet");
     }
 
-    return $ret if ($ret == 0 || $args{proceed_on_failure});
-    die("Background process $pid exit with $ret");
+    return 0;
```

Comment:
If you remove this, the error message could be misleading. We get `wasn't ready yet` also if the process exist with `!= 0`.
I think  you can extract the exit code with something like
```
my $ret = wait_serial "wait-DONE-\d+-";
unless(defined($ret)) {
   die(...)
}.....
($ret) = $ret =~ /wait-DONE-(\d+)-/;
return $ret if ($ret == 0 || $args{proceed_on_failure});
die("Background process $pid exit with $ret");
```

Reply:
Yes, It could be a problem, then let me move back the the original changes.